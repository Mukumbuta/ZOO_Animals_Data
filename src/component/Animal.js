import React, { useEffect, useState } from 'react';
import { Link } from 'react-router-dom';
import { useSelector, useDispatch } from 'react-redux';
import '../assets/homeStyles.css';
import acordionImg from '../assets/enter_icon.png';
import fetchAnimals from '../API/api';
import Header from './Header';

const Animal = () => {
  const animalList = useSelector((state) => state.animalsReducer);
  const [AnimalDetails, setAnimalDetails] = useState([]);
  const dispatch = useDispatch();

  useEffect(() => {
    dispatch(fetchAnimals());
  }, [dispatch]);

  return (
    <div className="animals-cont">
      <Header />
      <div className="search-bar-cont">
        <input
          id="search-input"
          type="text"
          placeholder="Search"
          onChange={(e) => setAnimalDetails(e.target.value)}
        />
      </div>
      <div id="home_content_cont">
        {animalList.length <= 1 ? <p>Just a moment please...</p> : (
          // eslint-disable-next-line
          animalList.filter((searchItem) => {
          // eslint-disable-next-line
          if (AnimalDetails == '') {
              return searchItem;
            } if (searchItem.name.toLowerCase().includes(AnimalDetails.toLowerCase())) {
              return searchItem;
            }
          }).map((animal) => (
            <Link key={animal.fish_id} className="content-cont" to={`/${animal.fish_id}`}>
              <div key={animal.fish_id}>
                <div className="animal_images">
                  <div className="img_accordion">
                    <img src={animal.actualimagelink} className="anim_img" alt={animal.name} />
                    <img src={acordionImg} className="accordion" alt="" />
                  </div>
                </div>
                <article className="heading">
                  <h1>
                    {animal.name}
                  </h1>
                </article>
              </div>
            </Link>
          ))
        )}
      </div>
    </div>
  );
};

export default Animal;
