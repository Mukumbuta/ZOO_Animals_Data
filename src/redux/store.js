import { createStore, combineReducers, applyMiddleware } from 'redux';
import logger from 'redux-logger';
import thunkMiddleware from 'redux-thunk';
import animalsReducer from './animals/animals';

const reducer = combineReducers({
  animalsReducer,
});

const store = createStore(
  reducer,
  applyMiddleware(logger, thunkMiddleware),
);

export default store;
