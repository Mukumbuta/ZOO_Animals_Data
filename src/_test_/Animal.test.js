import React from 'react';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import { BrowserRouter as Router } from 'react-router-dom';
import store from '../redux/store';
import Animal from '../component/Animal';

describe('Should render Animal component', () => {
  test('Should render HomePage', () => {
    const tree = renderer
      .create(<Router><Provider store={store}><Animal /></Provider></Router>)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
