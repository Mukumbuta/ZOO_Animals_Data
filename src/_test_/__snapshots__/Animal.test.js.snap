// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render Animal component Should render HomePage 1`] = `
<div
  className="animals-cont"
>
  <header
    className="header"
  >
    <img
      alt=""
      className="logo"
      src="zoo_animals.png"
    />
    <div
      className="heading"
    >
      <h1
        className="header_text"
      >
        ZOO Animals
      </h1>
      <p
        className="secText"
      >
        All animal types
      </p>
    </div>
  </header>
  <div
    className="search-bar-cont"
  >
    <input
      id="search-input"
      onChange={[Function]}
      placeholder="Search"
      type="text"
    />
  </div>
  <div
    id="home_content_cont"
  >
    <p>
      Just a moment please...
    </p>
  </div>
</div>
`;
